src/lib/libast/comp/iconv.c

```diff
@@ -523,20 +523,20 @@ if (error_info.trace < DEBUG_TRACE) sfprintf(sfstderr, "%s: debug%d: AHA#%d _ast
 		else if (!(cp = (const _ast_iconv_list_t*)ccmaplist((_ast_iconv_list_t*)cp)))
 			cp = codes;
 	}
-	if (cp == bp)
+	if (bp != 0)
```

Comment:
No. Do not make this transformation in a change whose sole purpose is eliminating a compiler warning about doing an assignment in a `if()` statement. Just hoist the assignment:

```
cp = bp;
if (cp)
```

Leave everything else alone. I don't care if you are confident this transformation is correct. The code is such a mess, and the unit tests so inadequate, that your elimination of the assignment to `cp` is too risky.

Reply:
Okay.